CodeChallenge/Base/CodeChallengeType.swift

```diff
@@ -27,17 +29,67 @@ protocol CodeChallengeType {
     
     /**
      A function to generate the dataset that the entries will be run against. Each `InputType` in the returned array will be run by each entry multiple times and then averaged in the results.
-    */
+     */
     func generateDataset() -> [InputType]
     
     /**
      A function which verifies the output from running an entry's block with the given input.
      
      - Returns: True if the output is valid, false if not.
-    */
+     */
     func verifyOutput(_ output: OutputType, forInput input: InputType) -> Bool
 }
 
+protocol JSONBasedChallenge: class, CodeChallengeType {
+    
+    associatedtype RawOutputType
+    
+    var fileName: String { get }
+    
+    var verificationData: [String: RawOutputType] { get set }
+    
+    func input(from raw: String) -> InputType
+    func raw(from input: InputType) -> String
+}
+
+extension JSONBasedChallenge {
+    
+    func generateDataset() -> [InputType] {
+        // Extract data from JSON file
+        let bundle = Bundle(for: BulletChallenge.self)
```

Comment:
This is just to get the 'current' bundle correct?
If so it might be better to add a private junk class to use in place of `BulletChallenge` here so this doesn't mysteriously break in the future?